src/common/config.ts

```diff
@@ -771,6 +775,21 @@ export class Config {
       : undefined;
   }
 
+  get preferredUnits(): 'metric' | 'imperial' {
+    return typeof this.settings.preferredUnits === 'string'
+      ? this.settings.preferredUnits
+      : 'metric';
```

Comment:
I guess the following would work:

```suggestion
    return this.settings.preferredUnits || 'metric';
```